src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -282,12 +282,12 @@ public void BuildCFIMap(ObjectNode node)
             _offsetToCfiEnd.Clear();
             _frameOpened = false;
 
-            if (!(node is INodeWithFrameInfo))
+            if (!(node is INodeWithCodeInfo))
```

Comment:
Fixed.

BTW: The assertion propagation in RyuJIT should be smart enough to remove the redundant cast.